api/api.go

```diff
@@ -21,19 +21,25 @@ type API interface {
 	CreateDatasetAPI(string, *mux.Router, store.DataStore) *DatasetAPI
 }
 
+// DownloadsGenerator pre generates full file downloads for the specified dataset/edition/version
+type DownloadsGenerator interface {
+	Generate(datasetID string, instanceID string, edition string, version string) error
```

Comment:
Do we need the instance ID passed in? usually its either the instanceID or the datasetID, edition, and version

Reply:
Its needed to get the observations in the data set exporter - I could remove it but then I would have to make additional calls to get it to do the filtering.